### ISSUE TITLE
First place scores on user profiles don't consider lazer scores

### ISSUE THREAD

**Issue by bdach** [ppy/osu-infrastructure#27]
~~In addition, the recent event section doesn't show when a user gets a high rank on a beatmap for lazer scores (https://github.com/ppy/osu-web/issues/11188).~~

- [ ] https://github.com/ppy/osu-web/pull/12337
- [ ] run `artisan beatmap-leaders:refresh` via web (will probably take a good while)
- [ ] https://github.com/ppy/osu-queue-score-statistics/pull/338
- [ ] https://github.com/ppy/osu-web/pull/12339

**Comment by bdach** [comment]
I made this an issue here because it seemed like the best place - feature is likely going to span multiple components.

The key question for me here is what we expect to happen, i.e. are "lazer mode off" rank no. 1s supposed to be accounted separately from "lazer mode on" rank no. 1s? The "no" answer simplifies implementation but kind of violates the "lazer mode off means lazer basically doesn't exist" premise; the "yes" answer addresses that concern but will also make implementation.... interesting?

**Comment by peppy** [comment]
To keep everyone happy and the toggle working, I'd propose that we just have a second table with lazer mode included in first place considerations, at least for an initial fix. This would also likely make implementation much simpler for us.

**Comment by bdach** [comment]
Well the current implementation uses four tables (`osu_leaders`, `osu_leaders_fruits`, `osu_leaders_mania`, `osu_leaders_taiko`), so would that mean four more? Or a new unified one? I think we could even get away with adding a `lazer` column to each of those...

**Comment by peppy** [comment]
Since we've been combining things until now, I'd probably lean towards a combined new table for storage.

**Comment by bdach** [comment]
@peppy given https://github.com/ppy/osu-web/issues/11188 has shipped, should I be prioritising this as a follow-up to it now?

**Comment by peppy** [comment]
Yeah, I think we can follow suit here and make it work for lazer leaderboards and only display with lazer mode enabled, avoiding issues that users have raised with getting this live.

It'll give us a solid implementation to migrate to whenever lazer mode goes away.
